### PR TITLE
fix: Fix itemgroup code and fix recharge scheme code

### LINKED ISSUE #5508: Picking up a grenade dropped by a monster causes detatched ptr error
### Describe the bug

Picking up grenades dropped by zombie soldiers and the like is causing the following error on highlighting it to pick up:
```
 DEBUG    : Attempted to resolve invalid detached_ptr

 FUNCTION : get
 FILE     : D:\a\Cataclysm-BN\Cataclysm-BN\src\detached_ptr.cpp
 LINE     : 57
 VERSION  : BN 606c81e
```

Oddly, grenades debugged in seem to be fine? I also tested the smoke bomb also dropped by a zombie soldier and no issue, just the grenade itself is being fucky.

### Steps To Reproduce

1. Spawn a grenade.
2. Drop it on the ground.
3. Use `g` to try and pick it up. Observe it works fine.
4. Spawn a bunch of zombie soldiers and debug-kill them.
5. Try to pick up a grenade off one of their bodies, it go error.

### Screenshots

Smoke bomb works fine...?
image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19041.450 (2004)
- Game Version: 606c81e [64-bit]
- Graphics Version: Tiles
- LAPI Version: 2
- Game Language:  []
- Mods loaded: [
    Bright Nights [bn]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by KheirFerrum**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

- fix #5508, due to #5283 which revealed the issue that our item group code will attempt to insert magazines into items that aren't explicitly labelled as having integral magazines, leading to null items being shoved into them and causing issues. Also fixes https://github.com/cataclysmbnteam/Cataclysm-BN/issues/5396
- fix #5283 where I noticed that my code will attempt to infinitely increment tool charges instead of respecting the maximum charges of a tool.

## Describe the solution

- item group item spawn code will now only load magazines into an item if it both doesn't already have one, and has a non-null item for a default magazine.

## Testing

- [x] Spawned a set of 9 zombie soldiers via debug menu, killed them all and prayed they spawned a grenade
  - [x] Picking up a grenade doesn't error the game, and it doesn't display it's contents as having "none"

## Additional context

Another shining testament to the spaghetti in our code. If I could eat all of our code spaghetti I'd never have to buy food again.

### CODE REVIEW

**Review by scarf005 — DISMISSED** [review]

**Comment by scarf005** [comment]
if so, could skip it with a comment explainging why it's needed:  https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics

File: src/relic.cpp [hunk 1]
```diff
@@ -82,8 +82,7 @@ void relic_recharge::load( const JsonObject &jo )
 {
     try {
         src_loc = jo.get_source_location();
-    } catch( const std::exception & ) {
-        // Savefiles don't specify source, so ignore error
+    } catch( const std::exception & ) { // NOLINT: Savefiles don't specify source, so ignore error
```
Comment (scarf005) [inline src/relic.cpp:85]: might be better to specify which error to ignore

**Review by scarf005 — DISMISSED** [review]

**Comment by chaosvolt** [comment]
Also fixes https://github.com/cataclysmbnteam/Cataclysm-BN/issues/5396 I assume.

**Comment by RoyalFox2140** [comment]
This needs testing on Windows.

**Review by chaosvolt — APPROVED** [review]
1. Compiled and load-tested.
2. Spawned a fuckload of zombie soldiers and debug-killed them.
3. Saw no sign of `none` in the grenades, picking them up no longer causes errors.
4. Spawned a bunch of regular zombies to repeat this, switchblades and other knives dropped show up as fine too.

**Pull request merged**